### PR TITLE
Support to pipeline

### PR DESCRIPTION
**Pull request by raphaelquati**
JENKINS-44068

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	<parent>
 		<groupId>org.jenkins-ci.plugins</groupId>
 		<artifactId>plugin</artifactId>
-		<version>1.532.3</version>
+		<version>1.626</version>
```
Comment (oleg-nenashev) [inline pom.xml]: I would recommend bumping to 1.642.x LTS baseline then, it's a good practice to test changes against Weekly releases. Update to Parent POM 2.x would be also helpfult, but it's likely out of the scope.

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java [hunk 2]
```diff
@@ -57,6 +65,62 @@ public ChangeAssemblyVersion(String versionPattern,
         this.assemblyTrademark = assemblyTrademark;
         this.assemblyCulture = assemblyCulture;
     }
+	
+    @DataBoundConstructor
+    public ChangeAssemblyVersion(String versionPattern) {
+        this.versionPattern = versionPattern;
+    }
+
+	@DataBoundSetter
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: tabs vs. spaces

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java [hunk 3]
```diff
@@ -170,13 +241,14 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
...
-    }
 
+            throw new AbortException(sw.toString());
+        }	    
+	    
+	}
+    
     @Extension
-    public static class Descriptor extends BuildStepDescriptor<Builder> {
+    public static class DescriptorImpl extends BuildStepDescriptor<Builder> {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: It formally breaks binary compatibility. Maybe it does not matter in the case of this plugin

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeTools.java [hunk 4]
```diff
@@ -26,7 +27,7 @@
         }
     }
 
-    public void Replace(String replacement, BuildListener listener) throws IOException, InterruptedException {
+    public void Replace(String replacement, TaskListener listener) throws IOException, InterruptedException {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeTools.java]: same, minor binary compat issue

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java (hunk 3 above)
Comment (raphaelquati) [reply src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: Without this rename, the plugin doesn't show in list in "Pipeline Syntax"

File: pom.xml (hunk 1 above)
Comment (raphaelquati) [reply pom.xml]: I'll change to 1.642.x

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java (hunk 3 above)
Comment (oleg-nenashev) [reply src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: @jglick @svanoort @abayer FYI, maybe some parsing logic specifics

**Comment by kdougl** [comment]
Any idea when this will be merged? Would really like to use this in a jenkins pipeline I'm working on at the moment?

**Comment by oleg-nenashev** [comment]
generally it needs feedback from the plugin maintainer (it's not me). If the plugin maintainer is not active, somebody needs to take its ownership.

P.S: I didn't get re-review request due to the force push. GitHub does not notify watchers in this case

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java [hunk 5]
```diff
@@ -118,16 +182,23 @@ public String getAssemblyCulture() {
      *
      */
     @Override
-    public boolean perform(AbstractBuild build, Launcher launcher, BuildListener listener) {
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: Probably ought to have a method matching the original signature that delegates to the new perform, to preserve binary compatibility.   Unless you can prove there are no plugins depending on this one.

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java [hunk 6]
```diff
@@ -170,13 +241,14 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
...
+            throw new AbortException(sw.toString());
         }
-        return true;
+        
     }
-
+    
     @Extension
-    public static class Descriptor extends BuildStepDescriptor<Builder> {
+    public static class DescriptorImpl extends BuildStepDescriptor<Builder> {
```
Comment (svanoort) [inline src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: Why rename?

**Comment by sgithubs** [comment]
It would be great to merge this feature 🥇

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java (hunk 6 above)
Comment (raphaelquati) [reply src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: Without this rename, the plugin doesn't show in list in "Pipeline Syntax"

Comment (raphaelquati) [reply src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: Without this rename, the plugin doesn't show in list in "Pipeline Syntax"

File: src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java (hunk 5 above)
Comment (raphaelquati) [reply src/main/java/org/jenkinsci/plugins/changeassemblyversion/ChangeAssemblyVersion.java]: New (old) method signature added.

**Comment by mcinnes01** [comment]
Any news on merging this?

**Comment by oleg-nenashev** [comment]
Somebody needs to take ownership of the plugin to release it.
I have moved to AppVeyor with my .NET projects, so I am not personally interested to take ownership (though I am happy to help to a new maintainer).

@jenkinsci/code-reviewers (esp. @slide) Does anybody have interest to take ownership?

**Comment by slide** [comment]
I can take it over, I'll file the PR for the permission repo

**Comment by sgithubs** [comment]
Hi, Do you have any instruction about pipeline syntax for such plugin?
Thanks.

**Comment by slide** [comment]
You can use the snippet generator to generate the code. Once it is released, there should be documentation automatically generated on jenkins.io.

**Comment by slide** [comment]
I had to upgrade a few things in the pom to get the release to work, but this should hit the update center within the next 24 hours.

**Comment by mtmargala** [comment]
What version of the plugin is this included in?  1.10 or 1.11? It looks like the update center got an update to 1.10 yesterday, but I didn't see this included.  I could be missing something.

**Comment by slide** [comment]
I looked at the source for the release and 1.10 has the pipeline support. What makes you think its not in there?

**Comment by slide** [comment]
FYI, I did tweak the support to allow a shortcut keyword in pipeline, instead of having to use the [step()] step, you can use changeAsmVer. If you use the snippet generator, it should show up there.

**Pull request merged**